### PR TITLE
cache footers; navbar by user and dashboard status

### PR DESCRIPTION
**Pull request by emilia-friedberg**
there is no specific issue, but this was in response to @amr-thameen's request to cache the navbar in order for the users not to see the Quill image "jump" when the page changes.

**Changes proposed in this pull request:**
- cache footers
- cache navbar (by current user and dashboard status)

**If this is a visual change please attach screencaps of the new branch, making sure to censor user data**

**Has this branch been QA'd on staging?** no

**Have you updated the docs?** no

**Have the tests been updated?** no

**Reviewer:** @ddmck

### CODE REVIEW

File: app/views/application/_sub_header.html.erb [hunk 1]
```diff
@@ -20,174 +20,176 @@
 	<% active_tab = '' %>
 <% end %>
 
-<nav class="q-navbar-home">
-	<div class="navbar-header">
-		<a class="navbar-brand" href=<%= root_path %>>
-			<img src="/images/quill_header_logo.svg" alt="Quill Logo">
-		</a>
-	</div>
+	<% cache "navbar-#{current_user&.id}-#{user_dropdown_class}" do %>
```
Comment (ddmck) [inline app/views/application/_sub_header.html.erb]: Why are we caching based on user id? Is it because we are displaying the username in the navbar?

**Comment by ddmck** [comment]
@emilia-friedberg In what way is the Quill image "jumping"? Because the navs aren't the same width?

**Comment by ddmck** [comment]
Where is the cache stored? Redis or in memory?

**Comment by ddmck** [comment]
This may be solvable by moving the loading of the css above the JS in the head partials. I think the jump is caused by some of the css being loaded by application-[chunkhash].css and some by app-bundle-[chunkhash].css, hence the first css styling loads, then the second, which causes the jump.

File: app/views/application/_sub_header.html.erb (hunk 1 above)
Comment (emilia-friedberg) [reply app/views/application/_sub_header.html.erb]: yes.

**Comment by emilia-friedberg** [comment]
@ddmck I believe the cache is stored in redis because that's the cache_store specified in our environment config files. I can try switching the order of the css and js in the head partials, but I think using the cache still has the potential to speed up page load times over the site. Is there a downside I'm not seeing?

**Pull request closed without merging**